src/jit/hwintrinsicxarch.cpp

```diff
@@ -1030,6 +1032,119 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_SSE2_SetAllVector128:
+        {
+            assert(sig->numArgs == 1);
+            baseType = getBaseTypeOfSIMDType(sig->retTypeSigClass);
+            assert((baseType >= TYP_BYTE) && (baseType <= TYP_DOUBLE));
+            assert(baseType != TYP_FLOAT);
+
+            op1          = impPopStack().val;
+            GenTree* src = nullptr;
+
+            switch (baseType)
+            {
+                case TYP_DOUBLE:
+                {
+                    src     = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_SSE2_SetScalarVector128, baseType, simdSize);
+                    retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, src, gtNewIconNode(0b01000100), NI_SSE2_Shuffle,
+                                                       TYP_INT, simdSize);
+                    break;
+                }
+
+                case TYP_LONG:
+                case TYP_ULONG:
+                {
+                    var_types shuffleBaseType = TYP_UNKNOWN;
+                    if (baseType == TYP_LONG)
+                    {
+                        intrinsic       = NI_SSE2_ConvertScalarToVector128Int64;
+                        shuffleBaseType = TYP_INT;
+                    }
+                    else
+                    {
+                        intrinsic       = NI_SSE2_ConvertScalarToVector128UInt64;
+                        shuffleBaseType = TYP_UINT;
+                    }
+#ifdef _TARGET_X86_
+                    src     = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_SSE2_SetScalarVector128, TYP_DOUBLE, simdSize);
+#else
+                    src     = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, intrinsic, baseType, simdSize);
+#endif
+                    retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, src, gtNewIconNode(0b01000100), NI_SSE2_Shuffle,
```

Comment:
Using `UnpackLow` (`punpcklqdq`) may be slightly more efficient.

Reply:
Actually I have followed MSVC compiler implementation and it seems to be quite efficient.

Reply:
Both `NI_SSE2_Shuffle` and `NI_SSE2_UnpackLow` are 1 cycle latency and 1 cycle throughput

Reply:
`UnpackLow` will result in slightly smaller codegen because it doesn't need to encode the immediate value.

Reply:
Right